RecoLocalTracker/SiPixelRecHits/src/PixelCPEClusterRepair.cc

```diff
@@ -91,6 +94,55 @@ PixelCPEClusterRepair::PixelCPEClusterRepair(edm::ParameterSet const & conf,
    maxSizeMismatchInY_ = conf.getParameter<double>("MaxSizeMismatchInY");
    minChargeRatio_ = conf.getParameter<double>("MinChargeRatio");
 
+   // read sub-detectors to recommend 2D
+   // can be: 
+   //     XYX (XYZ = PXB, PXE)
+   //     XYZ n (XYZ as above, n = layer, wheel or disk = 1 .. 6 ;)
+   std::vector<std::string> str_Recommend2D = conf.getParameter<std::vector<std::string> >("Recommend2D");
+   Recommend2D_.reserve(str_Recommend2D.size());
+   for (std::vector<std::string>::const_iterator it = str_Recommend2D.begin(), ed = str_Recommend2D.end(); it != ed; ++it) {
+     Recommend2D_.push_back(Rule(*it));
+   }
+
+   // run CR on damaged clusters (and not only on edge hits)
+   RunDamagedClusters_ = conf.existsAs<bool>("RunDamagedClusters")?conf.getParameter<bool>("RunDamagedClusters"):false;
+}
+
+//-----------------------------------------------------------------------------
+//  Fill all 2D template IDs
+//-----------------------------------------------------------------------------
+void PixelCPEClusterRepair::fill2DTemplIDs()
+{
+   auto const & dus = geom_.detUnits();
+   unsigned m_detectors = dus.size();
+   for(unsigned int i=1;i<7;++i) {
+      LogDebug("PixelCPEClusterRepair:: LookingForFirstStrip") << "Subdetector " << i
+      << " GeomDetEnumerator " << GeomDetEnumerators::tkDetEnum[i]
+      << " offset " << geom_.offsetDU(GeomDetEnumerators::tkDetEnum[i])
+      << " is it strip? " << (geom_.offsetDU(GeomDetEnumerators::tkDetEnum[i]) != dus.size() ?
+                              dus[geom_.offsetDU(GeomDetEnumerators::tkDetEnum[i])]->type().isTrackerStrip() : false);
+      if(geom_.offsetDU(GeomDetEnumerators::tkDetEnum[i]) != dus.size() &&
+         dus[geom_.offsetDU(GeomDetEnumerators::tkDetEnum[i])]->type().isTrackerStrip()) {
+         if(geom_.offsetDU(GeomDetEnumerators::tkDetEnum[i]) < m_detectors) m_detectors = geom_.offsetDU(GeomDetEnumerators::tkDetEnum[i]);
+      }
+   }
+   LogDebug("LookingForFirstStrip") << " Chosen offset: " << m_detectors;
+   
+   
+   m_DetParams.resize(m_detectors);
+   LogDebug("PixelCPEClusterRepair::fillDetParams():") <<"caching "<<m_detectors<<" pixel detectors"<<endl;
+   //Loop through detectors and store 2D template ID
+   bool printed_info = false;
+   for (unsigned i=0; i!=m_detectors;++i) {
+      auto & p=m_DetParams[i];
+
+      p.detTemplateId2D = templateDBobject2D_->getTemplateID(p.theDet->geographicalId());
+      if(p.detTemplateId != p.detTemplateId2D && !printed_info){
+          edm::LogInfo("PixelCPEClusterRepair") << "different template ID between 1D and 2D " << p.detTemplateId << " " << p.detTemplateId2D << endl;
```

Comment:
@OzAmram @cmantill 
I thought that after your last commit we would have found a message about the mismatch in the workflow steps which were failing before, but I cannot find any in:
   * [136.788 step3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26263/34038/runTheMatrix-results/136.788_RunSinglePh2017B+RunSinglePh2017B+HLTDR2_2017+RECODR2_2017reHLT_skimSinglePh_Prompt+HARVEST2017/step3_RunSinglePh2017B+RunSinglePh2017B+HLTDR2_2017+RECODR2_2017reHLT_skimSinglePh_Prompt+HARVEST2017.log)
   * [140.56 step2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26263/34038/runTheMatrix-results/140.56_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18/step2_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18.log)
   * [136.85 step3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26263/34038/runTheMatrix-results/136.85_RunEGamma2018A+RunEGamma2018A+HLTDR2_2018+RECODR2_2018reHLT_skimEGamma_Prompt_L1TEgDQM+HARVEST2018_L1TEgDQM/step3_RunEGamma2018A+RunEGamma2018A+HLTDR2_2018+RECODR2_2018reHLT_skimEGamma_Prompt_L1TEgDQM+HARVEST2018_L1TEgDQM.log)

shall we promote this to `LogWarning` instead?

Reply:
Hi Marco. Yes I think if we want them to be printed out we should promote it to LogWarning. When I tested locally, just running a runTheMatrix workflow did not print out the message, but adding a messageLogger to the config file did produce the message.

I will test right now that upgrading to LogWarning does print out a message for standard workflows

Reply:
Yes I can confirm that upgrading to LogWarning does print out a message in the standard workflow. So I suggest we do that